newrelic/api/datastore_trace.py

```diff
@@ -86,7 +87,8 @@ def __enter__(self):
             self.port_path_or_id = transaction._intern_string(self.port_path_or_id)
             self.database_name = transaction._intern_string(self.database_name)
 
-            datastore_tracer_settings = transaction.settings.datastore_tracer
+            settings = transaction.settings or global_settings()
+            datastore_tracer_settings = settings.datastore_tracer
```

Comment:
Do we want to be checking global_settings here or should we be following the pattern of `settings and datastore_tracer_settings.instance_reporting.enabled` so that the default if the transaction somehow has no settings is reading those settings as disabled?